internal/aws/ec2.go

```diff
@@ -272,6 +303,28 @@ func (c *Cloud) describeSecurityGroupsHelper(params *ec2.DescribeSecurityGroupsI
 	return results, err
 }
 
+// describeSubnetsHelper is a helper to handle pagination for DescribeSubnets API call
+func (c *Cloud) describeSubnetsHelper(params *ec2.DescribeSubnetsInput) (result []*ec2.Subnet, err error) {
+	err = c.ec2.DescribeSubnetsPages(params, func(output *ec2.DescribeSubnetsOutput, _ bool) bool {
+		result = append(result, output.Subnets...)
+		return true
+	})
+
+	return result, err
+}
+
+// hasClusterTagHelper is a helper to find if a Subnet has the correct ELB tag
+func (c *Cloud) hasClusterTagHelper(params []*ec2.Tag) bool {
+	for _, tag := range params {
```

Comment:
Hi, this logic should be:
1. If it's an internet-facing LB, looks for `kubernetes.io/role/elb` tag.
2. If it's an internal LB, looks for `kubernetes.io/role/internal-elb` tag.
Which aligns  with k8s, and provide an way to denote public/private subnet

Reply:
Thank you for the review,
I have updated the code with the new logic, I also updated the test case to handle this.